tests/test_connect.py

```diff
@@ -189,6 +219,22 @@ def tearDown(self):
             await self._try_connect(
                 user='md5_user', password='wrongpassword')
 
+    async def test_auth_password_scram_sha_256(self):
+        # scram is only supported in PostgreSQL 10 and above
+        if self.major_version < 10:
+            return
+
+        conn = await self.connect(
+            user='scram_sha_256_user', password='correctpassword')
```

Comment:
Please add more tests that check passwords which trigger SASLprep normalization, or fail SASLprep altogether.